<?xml version="1.0"?>
<launch>

    <!-- Name of the UWB sequence -->
    <arg name="bag_file" value="/home/kailai/Documents/100_dataset/util/util-uwb-dataset-main/dataset/flight-dataset/rosbag-data/const1/const1-trial1-tdoa2.bag"/>

    <!-- Launch the uwb imu fusion node -->
    <node pkg="ctgaupro" type="ctgaupro_gpui" name="ctgaupro_gpui" output="log"> <!--launch-prefix="xterm -e gdb -ex run -args"-->
        
        <param name="auto_exit"  value="1"/>

        <!-- Parameters for the Gaussian Process -->
        <param name="gpDt"       value="0.1"/>
        <param name="gpQr"       value="1.00"/>
        <param name="gpQc"       value="1.00"/>
        
        <!-- UWB anchor position -->
        <param name="anchor_pose_path" value="/home/kailai/Documents/100_dataset/util/util-uwb-dataset-main/dataset/flight-dataset/survey-results/anchor_const1_survey.txt"/>
        <param name="tdoa_topic" value="/tdoa_data"/>
        <param name="tof_topic"  value="/tof_data" />
        <param name="imu_topic"  value="/imu_data" />
        <param name="gt_topic"   value="/pose_data" />
        <param name="fuse_tdoa"  value="1"/>
        <param name="fuse_tof"   value="1"/>
        <param name="fuse_imu"   value="1"/>
        <param name="auto_diff"  value="0"/>

        <!-- Parameters for the solver  -->
        <param name="SLIDE_SIZE"  value="1"/> <!-- How many knots to slide -->
        <param name="WINDOW_SIZE" value="10"/> <!-- How many knot length does the sliding window -->
        <param name="w_tdoa" value="100.0"/> <!-- Coefficients for TDOA residuals -->
        <param name="GYR_N" value="2000.0"/> <!-- Coefficients for IMU residuals -->
        <param name="GYR_W" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="ACC_N" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="ACC_W" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="tdoa_loss_thres" value="30.0"/> <!-- Loss function for IMU residuals -->
        <param name="mp_loss_thres" value="100.0"/> <!-- Loss function for motion prior residuals -->

    </node>

    <!-- Play the bag file -->
    <node required="false" pkg="rosbag" type="play" name="rosbag_player"
          args="--clock $(arg bag_file) -r 1.0 -s 0.0"
          launch-prefix="bash -c 'sleep 1.0; $0 $@' "/>

    <!-- Visualize -->
    <node pkg="rviz" type="rviz" name="rviz_ui" output="log" required="true"
        args="-d $(find ctgaupro)/launch/ctgaupro_ui.rviz" />          

</launch>